core/src/engine/io/fd_control.hpp

```diff
@@ -75,10 +83,12 @@ class Direction final {
 
   static void IoWatcherCb(struct ev_loop*, ev_io*, int) noexcept;
 
+  bool is_awaitable_{true};
   int fd_{-1};
   const Kind kind_;
   std::atomic<bool> is_valid_;
   Mutex mutex_;
+  std::mutex blocking_mutex_;
```

Comment:
Yesyerday we had a disscussion on Socket design. We'll try to remove all the mutexes from socket in the next few days. No 100% gurantee that this will happen.

Reply:
We made it. No more mutexes in socket https://github.com/userver-framework/userver/commit/1092847cf0443b251b215dc5576fe6148a4f5d17